langchain4j/src/test/java/dev/langchain4j/service/ServiceOutputParserTest.java

```diff
@@ -203,4 +203,27 @@ void outputFormatInstructions_PersonWithFinalFields() {
                         "\"birthDate\": (type: date string (2023-12-31)),\n" +
                         "}");
     }
+
+    static class PersonWithParents {
+        private String firstName;
+        private String lastName;
+        private List<PersonWithParents> parents;
+    }
+
+    @Test
+    void outputFormatInstructions_PersonWithParents() {
+        String formatInstructions = ServiceOutputParser.outputFormatInstructions(PersonWithParents.class);
+
+        assertThat(formatInstructions).isEqualTo(
+                "\nYou must answer strictly in the following JSON format: {\n" +
+                        "\"firstName\": (type: string),\n" +
+                        "\"lastName\": (type: string),\n" +
+                        "\"parents\": (type: array of dev.langchain4j.service.ServiceOutputParserTest$PersonWithParents: {\n" +
```

Comment:
Instead of printing PersonWithParents fields twice, should it just put "dev.langchain4j.service.ServiceOutputParserTest$PersonWithParents" before the first description and then just reference itself? This will potentially save a lot of tokens.

Something like this (pseudo-JSON):
```
PersonWithParents:
 - firstName
 - lastName
 - parents: array of PersonWithParents
```

Reply:
I think it's ok.